server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -674,7 +678,15 @@ public void setLastAcceptedState(ClusterState clusterState) {
             try {
                 final ClusterMetadataManifest manifest;
                 if (shouldWriteFullClusterState(clusterState)) {
-                    manifest = remoteClusterStateService.writeFullMetadata(clusterState);
+                    if (clusterState.metadata().clusterUUIDCommitted() == true) {
```

Comment:
so this means an invalid clusterUUID would never be linked to a previous valid cluster uuid?

Reply:
The previousClusterUUID is being set in the constructor itself. When clusterUUIDCommitted is true, previous cluster UUID will be used using the value set in the constructor.
Here we are fetching the previousUUID from the latest manifest in case this master was elected newly so that to handle the case stale UUID set in the constructor.

Reply:
2 cases i can think of where new cluster manager would be elected:
1. Failover from active cluster manager - in this case if previousClusterUUID in constructor is stale, that means another clusterUUID became valid during failover. This doesnt sound like a valid case either
2. Cluster bootstrap/restart/recovery - in this case you would already have the correct previousClusterUUID because you just fetched the previousClusterUUID during auto restore if there existed one.

Any specific example where previousClusterUUID in constructor would be stale

Reply:
Think of this case - Cluster is working fine with M1, M2 and M3 nodes on UUID1. All nodes go down and M4,M5 and M6 come up with previous UUID as UUID1. Then M4 and M5 form a cluster on UUID2. M6 has come up but not joined due to some network partitioning. The M4 and M5 go down and M7 and M8 come up. M7 and M8 are seeded correctly to form cluster on UUID3 with previous UUID as UUID2. Now M6 joins the cluster but it still has previous UUID as UUID1. If the failover of current cluster manager occurs and M6 becomes active, it will have stale previous cluster UUID.